src/query.c

```diff
@@ -1753,63 +1776,91 @@ int QueryNode_ForEach(QueryNode *q, QueryNode_ForEachCallback callback, void *ct
   return retVal;
 }
 
+// Convert the query attribute into a raw vector param to be resolved by the vector iterator
+// down the road. return 0 in case of an unrecognized parameter.
+static int QueryVectorNode_ApplyAttribute(VectorQuery *vq, QueryAttribute *attr) {
+  if (STR_EQCASE(attr->name, attr->namelen, VECSIM_EFRUNTIME) ||
+      STR_EQCASE(attr->name, attr->namelen, VECSIM_HYBRID_POLICY) ||
+      STR_EQCASE(attr->name, attr->namelen, VECSIM_BATCH_SIZE)) {
+    VecSimRawParam param = (VecSimRawParam){ .name = rm_strndup(attr->name, attr->namelen),
```

Comment:
Why do we need strdup here? The raw param is `const char *`

Reply:
The caller function `QueryNode_ApplyAttributes` is called from the parser, where `QueryAttribute` struct is being constructed and **destructed**. Hence, we need to move the ownership for these string here. But you're right that we can just set `attr->name` and `attr->value` to be NULL instead of using `strndup`.